### PR TITLE
[New Rule] GitHub Owner Role Granted to User

### PR DESCRIPTION
**Pull request by imays11**
## Issues

https://github.com/elastic/detection-rules/issues/3084 

## Summary
This rule detects when a member is granted the organization owner role of a GitHub organization. 
This role provides admin level privileges. Any new owner roles should be investigated to determine it's validity. 
Unauthorized owner roles could indicate compromise within your organization and provide unlimited access to data and settings.

### CODE REVIEW

File: rules/integrations/github/persistence_organization_owner_role_granted.toml [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+license = "Elastic License v2"
+name = "GitHub Owner Role Granted To User"
+risk_score = 47
+rule_id = "9b343b62-d173-4cfd-bd8b-e6379f964ca4"
+severity = "medium"
+tags = ["Domain: Cloud", "Use Case: Threat Detection", "Tactic: Persistence", "Data Source: Github"]
+timestamp_override = "event.ingested"
+type = "eql"
+query = '''
+iam where event.dataset== "github.audit" and event.action== "org.update_member" and github.permission== "admin"
```
Comment (Aegrah) [inline rules/integrations/github/persistence_organization_owner_role_granted.toml:28]:
```suggestion
iam where event.dataset == "github.audit" and event.action == "org.update_member" and github.permission == "admin"
```

File: rules/integrations/github/persistence_organization_owner_role_granted.toml [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+maturity = "production"
+min_stack_comments = "New fields added: required_fields, related_integrations, setup"
+min_stack_version = "8.3.0"
+updated_date = "2023/09/11"
+
+[rule]
+author = ["Elastic"]
+description = """
+This rule detects when a member is granted the organization owner role of a GitHub organization. 
+This role provides admin level privileges. Any new owner roles should be investigated to determine it's validity.
```
Comment (Aegrah) [inline rules/integrations/github/persistence_organization_owner_role_granted.toml:13]:
```suggestion
This role provides admin level privileges. Any new owner role should be investigated to determine its validity. 
```

**Review by Aegrah — APPROVED** [review]
LGTM after above nits.

**Review by brokensound77 — APPROVED** [review]
LGTM with @Aegrah suggestions

**Review by terrancedejesus — APPROVED** [review]

**Pull request merged**